src/Illuminate/Database/Eloquent/Collection.php

```diff
@@ -139,6 +139,27 @@ public function map(callable $callback)
     }
 
     /**
+     * Reload a fresh model instance from the database for all the entities.
+     *
+     * @param  array|string  $with
+     * @return static
+     */
+    public function fresh($with = [])
+    {
+        $model = $this->first();
```

Comment:
I have no code. I just assume if `Collection` is empty this line returns null and next code will fail with `Trying to get property of non-object`